doc/map-reduce.md

```diff
@@ -1,69 +1,123 @@
 # Map/reduce pattern
 
-The **map/reduce** pattern combines a **map** and **reduce** operation in a single pattern. Firstly one or more input data sets are *mapped* by applying a transformation operation. Then, the results are combined by means of a reduction operation.
+The **map/reduce** pattern combines a **map** and **reduce** operation in a
+single pattern. In a **map/reduce** operation, firstly one or more input data
+sets are *mapped* by applying a transformation operation. Then, the results are
+combined by means of a reduction operation.
 
-The interface to the **mapreduce** pattern is provided by function `grppi::map_reduce()`. As all functions in *GrPPI*, this function takes as its first argument an execution policy.
+The interface to the **map/reduce** pattern is provided by function
+`grppi::map_reduce()`. As all functions in *GrPPI*, this function takes as its
+first argument an execution policy.
 
 ~~~{.cpp}
 grppi::map_reduce(exec, other_arguments...);
 ~~~
 
-**Note**: A **map/reduce** could be also expressed by the composition of a **map** and a **reduce**. However, **map/reduce** fuses both stages allowing for extra optimizations.
+**Note**: A **map/reduce** could be also expressed by the composition of a
+**map** and a **reduce**. However, **map/reduce** fuses both stages allowing for
+extra optimizations.
```

Comment:
Rephrased to:

However, **map/reduce** may fuse both stages, allowing for extra optimizations.